client/utilities/rfc.ts

```diff
@@ -0,0 +1,124 @@
+import type { RfcMetadata, Rfc } from '../generated/red-client'
+import { NONBREAKING_SPACE } from './strings'
+
+const cache: Record<string, Rfc> = {}
+
+export const rfcMetadataToRfc = (rfcMetadata: RfcMetadata): Rfc => {
+  const rfcId = parseRFCId(`rfc${rfcMetadata.number}`)
+  const cacheId = `${rfcId.type}${rfcId.number}`
+  if (!cache[cacheId]) {
+    cache[cacheId] = {
+      ...blankRfc,
+      number: rfcMetadata.number,
+      abstract: rfcMetadata.abstract,
+      published: rfcMetadata.published,
+      status: rfcMetadata.status,
+      pages: rfcMetadata.pages,
+      authors: rfcMetadata.authors,
+      group: rfcMetadata.group,
+      area: rfcMetadata.area,
+      stream: rfcMetadata.stream,
+      identifiers: rfcMetadata.identifiers,
+      obsoleted_by: rfcMetadata.obsoleted_by,
+      updated_by: rfcMetadata.updated_by,
+      title: rfcMetadata.title
+    }
+  }
+  return cache[cacheId]
+}
+
+export const blankRfc: Rfc = {
+  number: 0,
+  title: '',
+  published: '1950-1-1',
+  status: {
+    // @ts-expect-error waiting on fixes to Red via api client
+    slug: 'standard',
+    name: 'standards track'
+  },
+  pages: 0,
+  authors: [],
+  group: {
+    acronym: '',
+    name: ''
+  },
+  area: {
+    acronym: '',
+    name: ''
+  },
+  stream: {
+    slug: '',
+    name: '',
+    desc: ''
+  },
+  // @ts-expect-error waiting on fixes to Red via api client
```

Comment:
I think this is the error you mentioned on slack. What is the error? Earlier in this PR, it looks like the generated code for `RfcMetadata` uses a `DocIdentifier` that has the correct structure.

Oh, looks like it has a single `DocIdentifier` rather than a list of them. Is that it? (The code returns an array, but the schema incorrectly shows it as a single object)

Reply:
Yes I think that's about it.

The nice thing about `@ts-expect-error` vs `@ts-ignore` is that when the API client's types are fixed TS will make me remove this directive, rather than allowing me to leave a dangerous wildcard 'ignore all TS errors' in the code